src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -932,6 +935,19 @@ def causal_mask_input_present(model_path: str) -> bool:
             for inp in onnx.load(model_path, load_external_data=False).graph.input
         )
 
+    @contextmanager
```

Comment:
let's make a `ChatPipeline(TextGenerationPipeline)` that adds this functionality